src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -692,6 +692,18 @@ internal bool HandleMouseRelease(object sender, MouseButtonEventArgs e)
 
                     if (dropGroup != null)
                     {
+                        // If groups are being dragged store them here
+                        var dragedGroups = DynamoSelection.Instance.Selection
+                            .OfType<AnnotationModel>()
+                            .ToList();
+
+                        // We do not want to add dragged groups content twice
+                        // so we filter it out here.
+                        var modelsToAdd = DynamoSelection.Instance.Selection
+                            .OfType<ModelBase>()
+                            .Except(dragedGroups.SelectMany(x => x.Nodes))
```

Comment:
By reading this code, seems annotation within the dragged group could be linked to the hovered-over group with this approach?

Reply:
@QilongTang not sure i understand correctly?

Reply:
Ah, we already are. `Nodes` is everything that a Group contains (notes, nodes and groups) its just poorly named

Reply:
ohh, that's right.. Thanks, then this looks good